app/models/plan.rb

```diff
@@ -725,6 +749,15 @@ def visibility_allowed?
 
   private
 
+  # Returns whether or not the user has the specified role for the plan
+  def has_role(user_id, role_as_sym)
+    if user_id.is_a?(Integer) && role_as_sym.is_a?(Symbol)
```

Comment:
be aware that checking types is an expensive operation, if this method is used only within this class, I would make it private and remove the type checking